addons/account/models/chart_template.py

```diff
@@ -466,31 +466,44 @@ def deref(values, model):
 
         def defer(all_data):
             """Defer writing some relations if the related records don't exist yet."""
+
+            def should_defer(created_models, yet_to_be_created_models, model, field_name, field_val):
+                def can_defer(created_models, yet_to_be_created_models, model, field):
+                    return (field.comodel_name not in created_models
+                            and (field.comodel_name == model
+                                 or field.comodel_name in yet_to_be_created_models))
+
+                field = self.env[model]._fields.get(field_name, None)
+                if not field or not field.relational or not field_val:
+                    return False
+
+                if isinstance(field_val, list | tuple) and isinstance(field_val[0], list | tuple):
+                    # If it's a command
+                    for command, *rest_of_tuple in field_val:
+                        if command == Command.CLEAR:
+                            continue
+                        if command != Command.CREATE:
```

Comment:
These `if`s are hard to read: `if: continue` then `if not: ; else:`

Pattern matching would make this easier to understand I think

It would also allow to have a proper name for that dirty "`rest_of_tuple`"...

Reply:
![image](https://github.com/odoo/odoo/assets/1665365/fd602387-a030-4d28-9a0e-0f0ea69cd718)

Reply:
It looks like the CI actually cares, and `ruff` is still long to come. See: https://discord.com/channels/678381219515465750/704997651833159831/1220392497541419066

Reply:
Pylint can be overridden, as long as ruff is happy and we can justify why we override.

Reply:
That's called cheating.
Re-instating then.